src/main/java/net/openhft/chronicle/threads/MediumEventLoop.java

```diff
@@ -54,6 +54,10 @@ public class MediumEventLoop extends AbstractLifecycleEventLoop implements CoreE
     protected static final EventHandler[] NO_EVENT_HANDLERS = {};
     protected static final long FINISHED = Long.MAX_VALUE - 1;
 
+    private final Object startStopMutex = new Object();
+    private final Object setHighHandlerMutex = new Object();
+    private final Object copyMediumArrayMutex = new Object();
```

Comment:
We could make these `Lock`s if we decided we needed the extra API, I think the intrinsic lock is sufficient here though.